### ISSUE TITLE
seems the ringbuffer is not thread safe

### ISSUE THREAD

**Issue by lijin1989422** [apppur/cocos2dx-network#1]
seems the member variables **m_tail** and **m_head** need to use some wapper to enhance for the mutilthread env to prevent them to be read and wrote by diffrent threads in the same time.

**Comment by apppur** [comment]
yes, there is a problem for many reader and writer. only this for one reader and one writer.
